src/scripts/sniperino.ts

```diff
@@ -0,0 +1,101 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    getNewSnipe();
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
+    
+    constructor(n: string) {
+        this.name = n.slice(0, n.indexOf(" "));
+        this.gamesPlayed = 0;
+        this.gamesWon = 0;
+        //this.currentSnipe = this.getNewSnipe();
+    }
+
+    public getWinRate(): number {
+        return this.gamesPlayed > 0 ? (this.gamesWon / this.gamesPlayed) * 100 : 0;
+    }
+
+    public getNewSnipe() {
+        this.currentSnipe = Math.floor(Math.random() * 99) + 1;
+    }
+}
+
+let game = (robot: Robot) => {
+
+    // Get a sniper with a given name. If one doesn't exist, create it.
+    let getSniper = (name: string) => {
```

Comment:
Maybe instead make a new class which takes a Robot as a constructor param and have a Start method like:

this.robot.respond(/sniperino.../, this.startSniperino)
this.robot.on("roll", this.onRoll)

this class could have a getSniper function as defined

then the exported function would just be: (robot: Robot) => { new SniperinoListener(robot).Start(); }

idk